tools/python_api/test/test_datatype.py

```diff
@@ -393,3 +396,22 @@ def test_rdf_variant(conn_db_readwrite: ConnDB) -> None:
 
     with conn.execute("DROP RDFGraph T;") as result:
         result.close()
+
+
+def test_large_array(conn_db_readwrite: ConnDB) -> None:
+    conn, db = conn_db_readwrite
+
+    data = []
+    for i in range(1000):
+        data.append({
+            "id": i,
+            "embedding": np.random.rand(1670).tolist()
+        })
+
+    df = pd.DataFrame(data)
```

Comment:
I think the same test should also be done for PyArrow/Polars too, what do you think @mxwli ?